x-pack/plugins/apm/server/lib/settings/custom_link/create_custom_link_index.ts

```diff
@@ -39,20 +39,20 @@ const mappings: Mappings = {
     service: {
       properties: {
         name: {
-          type: 'keyword'
+          type: 'text'
         },
         environment: {
-          type: 'keyword'
+          type: 'text'
         }
       }
     },
     transaction: {
       properties: {
         name: {
-          type: 'keyword'
+          type: 'text'
         },
         type: {
-          type: 'keyword'
+          type: 'text'
```

Comment:
This looks wrong. None of these fields should be of type `text`

Reply:
text are for terms that could "sort of" match and therefore gets a score assigned where keywords are boolean: either a phrase matches (exactly) or it doesn't. We want the latter behaviour. Using text could cause links to show up in the wrong places under certain conditions (if service names are similarly named)

Reply:
I also thought, but check what the subtitle of the filter section says: `Add additional values within the same field by comma separating values.` It means the user can select `service.name`:`opbeans-java, opbeans-node` and this link should be shown on both services.

Reply:
This also changed the query, from `term` to `match` https://github.com/elastic/kibana/pull/59648/files/0f8deb492fb59dcfba134e299b9dd982c525237d#diff-e3f06dd0efa093de35afee906d872a3bR25

Reply:
> It means the user can select service.name:opbeans-java, opbeans-node and this link should be shown on both services.

Sure, then we should add those as arrays.